plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/CreateMissingFBMarkerResolution.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.marker.resolution;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.commands.change.UpdateFBTypeCommand;
+import org.eclipse.fordiac.ide.model.commands.change.UpdateInternalFBCommand;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.BaseFBType;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.fordiac.ide.model.libraryElement.FB;
+import org.eclipse.fordiac.ide.model.search.AbstractLiveSearchContext;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.impl.ErrorFBTypeEntryImpl;
+import org.eclipse.fordiac.ide.typemanagement.wizards.RestrictedNewTypeWizard;
+import org.eclipse.fordiac.ide.ui.FordiacMessages;
+import org.eclipse.ui.PlatformUI;
+
+public class CreateMissingFBMarkerResolution extends AbstractErrorMarkerResolution {
+
+	private TypeEntry newEntry;
+	List<EObject> targets = new ArrayList<>();
+
+	public CreateMissingFBMarkerResolution(final IMarker marker) {
+		super(FordiacMessages.Repair_Dialog_New_FB, marker);
+	}
+
+	@Override
+	public void run(final IMarker marker) {
+		final EObject target = FordiacErrorMarker.getTarget(marker);
+		updateTarget(target);
+	}
+
+	private void updateTarget(final EObject target) {
+		if (target instanceof final ErrorMarkerFBNElement err) {
+			AbstractLiveSearchContext.executeAndSave(new UpdateFBTypeCommand(err, newEntry), err,
+					new NullProgressMonitor());
+		}
+
+		if (target instanceof final FB fb && fb.getTypeEntry() instanceof ErrorFBTypeEntryImpl
+				&& fb.eContainer() instanceof final BaseFBType base && base.getInternalFbs().contains(fb)) {
+			AbstractLiveSearchContext.executeAndSave(new UpdateInternalFBCommand(fb, newEntry), fb,
+					new NullProgressMonitor());
+		}
+	}
+
+	@Override
+	public void run(final IMarker[] markers, final IProgressMonitor monitor) {
```

Comment:
Looking at the code, it seems we have no choice but to process all markers at once here. In that case, I would suggest to delegate from `run(IMarker)` to this method instead to handle the case where that method is called directly. Please have a look at `WorkbenchMarkerResolutionAdapter.run(IMarker)` for an example.